sql/handler.cc

```diff
@@ -1910,6 +1910,8 @@ int ha_commit_trans(THD *thd, bool all)
     if (run_wsrep_hooks && !error)
       error= wsrep_after_commit(thd, all) ? 2 : 0;
 #endif /* WITH_WSREP */
+    if (error)
+      goto err;
```

Comment:
This seems to be wrong, a failure of `wsrep_after_commit()` does not mean a failure of the transaction commit.